ionoscloud/utils.go

```diff
@@ -29,24 +30,33 @@ func responseBody(resp *ionoscloud.APIResponse) string {
 }
 
 // DiffBasedOnVersion used for k8 node pool and cluster
+// ignores downgrades of the patch versions.
 func DiffBasedOnVersion(_, old, new string, _ *schema.ResourceData) bool {
 	var oldMajor, oldMinor string
 	var newMajor, newMinor string
+	var oldPatchInt, newPatchInt int
 	if old != "" {
 		oldSplit := strings.Split(old, ".")
-		if len(oldSplit) > 1 {
+		if len(oldSplit) > 2 {
```

Comment:
@cristiGuranIonos not sure why we need this length check since the version should always have the format `x.y.z`, meaning that the length will always be 3.

Anyway, now we have `> 2` instead of `> 1`, so we are restricting the number of cases in which we enter the condition. There are two things unclear for me:

1. Why do we need a length check?
2. Why did we increase the length in the check since we have the same format for version?

Reply:
To make sure that the split doesn't panic and we don't read above the array boundary.  Teoretically a user might do something like 1.20 although not supported